### PR TITLE
#42 draft of ajax autocomplete

### PR DESCRIPTION
**Pull request by 4n70w4**
This is a draft. Opinions and comments are welcome!

I did not quite understand the original plan of the author,
so it turned out chaotically and requires improvements.

But you can already use it!

### CODE REVIEW

File: src/Http/Controllers/AttachController.php [hunk 1]
```diff
@@ -18,15 +18,21 @@ public function create(NovaRequest $request, $parent, $relationship)
...
     }
 
-    public function getAvailableResources($request, $relationship)
+    public function getAvailableResources(NovaRequest $request, $relationship)
     {
         $resourceClass = $request->newResource();
 
+        $search = trim($request->query('search'));
+
+        if(empty($search) ) {
```
[the remark this answers is not in the document]
Comment (4n70w4) [reply src/Http/Controllers/AttachController.php]: Thank you for a useless comment)

File: src/Http/Controllers/AttachController.php [hunk 2]
```diff
@@ -36,6 +42,31 @@ public function getAvailableResources($request, $relationship)
         $query = $field->resourceClass::newModel();
 
         return $field->resourceClass::relatableQuery($request, $query)->get()
+            ->mapInto($field->resourceClass)
+            ->filter(function ($resource) use ($request, $field, $search) {
+                return $request->newResource()->authorizedToAttach($request, $resource->resource) && false !== mb_stripos($resource->title(), $search);
+            })->map(function($resource) {
```
[the remark this answers is not in the document]
Comment (4n70w4) [reply src/Http/Controllers/AttachController.php]: Thank you for a useless comment)

File: src/Http/Controllers/AttachController.php [hunk 3]
```diff
@@ -44,6 +75,7 @@ public function getAvailableResources($request, $relationship)
                     'display' => $resource->title(),
                     'value' => $resource->getKey(),
                 ];
-            })->sortBy('display')->values();
+            })->sortBy('display')->slice(0, 10)->values();
     }
-}
\ No newline at end of file
+
```
[the remark this answers is not in the document]
Comment (4n70w4) [reply src/Http/Controllers/AttachController.php]: Thank you for a useless comment)

**Comment by dillingham** [comment]
Sorry but this isn't the intention of this package.
Glad you were able to make it work for your usecase.
Kind of weird that you published a fork of it on novapackages though

**Comment by 4n70w4** [comment]
@dillingham well, ok, I will develop it as an independent product.

**Comment by niladam** [comment]
On a second thought, @dilingham - I think ajax autocomplete does make sense when using a huge load of options. Having this in the package, even if requires activation somehow does seem like a good option.

Care to elaborate on the intention you're talking about? Also, maybe provide some reasons as to why this wouldn't make the package better?

**Pull request closed without merging**